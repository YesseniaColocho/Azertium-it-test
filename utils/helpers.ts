export function newURL(id: number): string{
    return `https://via.assets.so/album.png?id=${id}&q=100&w=600&h=600&fit=fill`
}