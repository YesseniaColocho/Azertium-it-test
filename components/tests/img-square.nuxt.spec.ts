import { describe, it, expect, vi} from 'vitest'
import { shallowMount } from '@vue/test-utils'
import * as helpers from '~/utils/helpers'

import imgSquare from '../img-square.vue'

const imageProp = { id: 1 }

describe('image square', () => {

    const spyNewURL = vi.spyOn(helpers, 'newURL').mockImplementation(() => 'https://test.com/image.png')

    const component = () => shallowMount(
      imgSquare, 
      {
        props: {
          imageProp: imageProp
        }
      }
    )

    it('component renders image square properly', () => {
      const wrapper = component()
      expect(wrapper).toBeTruthy()
    })

    it('component html renders properly after setting tagImage variable', async () => {
      const wrapper = component()
      expect(spyNewURL).toHaveBeenCalledWith(1)
      wrapper.vm.setTagImage('https://test.com/image.png')
      await wrapper.vm.$nextTick();
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('Test hideImg', async () => {
      const wrapper = component()
      expect(spyNewURL).toHaveBeenCalledWith(1)
      const imageWrapper = wrapper.find('.image');
      await imageWrapper.trigger('click');
      await wrapper.vm.$nextTick();
      expect(wrapper.find('.image').exists()).toBe(false);
    })

})