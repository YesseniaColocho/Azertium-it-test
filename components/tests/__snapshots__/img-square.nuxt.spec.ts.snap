// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`image square > component html renders properly after setting tagImage variable 1`] = `
"<transition-stub data-v-dfd8a8bd="" appear="false" persisted="false" css="true">
  <div data-v-dfd8a8bd="" class="image">
    <transition-stub data-v-dfd8a8bd="" appear="false" persisted="false" css="true"><img data-v-dfd8a8bd="" src="https://test.com/image.png"></transition-stub>
  </div>
</transition-stub>"
`;
